pkg/apis/logicalcluster/v1alpha1/logical-cluster.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// LogicalCluster represents a cluster.
+//
+// +crd
+// +genclient
+// +genclient:nonNamespaced
```

Comment:
I am not sure this is the right choice.
I am not sure it is not.

Reply:
I think it's mainly automatic writing :-)  I don't have strong opinion here - what do you think?

Reply:
I think that in general a resource should NOT be namespaced iff there is a specific reason why it logically applies to the whole of the cluster wherein it appears.  Otherwise, why take away the ability of users to organize the resource into namespaces?

Reply:
Changed logicalcluster to be namespaced. We plan to hot logicalcluster objects in a NS corresponding to provider instance.

Reply:
pushed in new commiy